cli/generate_operator_keys.go

```diff
@@ -1,123 +1,125 @@
-		}
-
-		// Resolve to absolute path
-		passwordAbsPath, err := filepath.Abs(passwordFilePath)
-		if err != nil {
-			logger.Fatal("Failed to read absolute path of password file", zap.Error(err))
-		}
 
-		// Now read the file
-		// #nosec G304
-		passwordBytes, err := os.ReadFile(passwordAbsPath)
+		pk, sk, err := rsaencryption.GenerateKeys()
 		if err != nil {
-			logger.Fatal("Failed to read password file", zap.Error(err))
+			logger.Fatal("Failed to generate keys", zap.Error(err))
 		}
 
-		encryptionPassword := string(passwordBytes)
-
 		if privateKeyFilePath != "" {
-			// Resolve to absolute path
-			privateKeyAbsPath, err := filepath.Abs(privateKeyFilePath)
+			sk, pk, err = readPrivateKeyFromFile(privateKeyFilePath, logger)
 			if err != nil {
-				logger.Fatal("Failed to read absolute path of private key file", zap.Error(err))
-			}
-
-			// Now read the file
-			// #nosec G304
-			privateKeyBytes, _ := os.ReadFile(privateKeyAbsPath)
-			if privateKeyBytes != nil {
-				keyBytes, err := base64.StdEncoding.DecodeString(string(privateKeyBytes))
-				if err != nil {
-					logger.Fatal("base64 decoding failed", zap.Error(err))
-				}
-
-				keyPem, _ := pem.Decode(keyBytes)
-				if keyPem == nil {
-					logger.Fatal("failed to decode PEM", zap.Error(err))
-				}
-
-				rsaKey, err := x509.ParsePKCS1PrivateKey(keyPem.Bytes)
-				if err != nil {
-					logger.Fatal("failed to parse RSA private key", zap.Error(err))
-				}
-
-				skPem := pem.EncodeToMemory(
-					&pem.Block{
-						Type:  "RSA PRIVATE KEY",
-						Bytes: x509.MarshalPKCS1PrivateKey(rsaKey),
-					},
-				)
-
-				operatorPublicKey, _ := rsaencryption.ExtractPublicKey(rsaKey)
-				publicKey, _ := base64.StdEncoding.DecodeString(operatorPublicKey)
-				sk = skPem
-				pk = publicKey
+				logger.Fatal("Failed to read private key from file", zap.Error(err))
 			}
 		}
 
 		if err := logging.SetGlobalLogger("debug", "capital", "console", nil); err != nil {
 			logger.Fatal("", zap.Error(err))
 		}
 
-		if err != nil {
-			logger.Fatal("Failed to generate operator keys", zap.Error(err))
-		}
-		logger.Info("generated public key (base64)", zap.String("pk", base64.StdEncoding.EncodeToString(pk)))
-
-		if encryptionPassword != "" {
-			encryptedData, err := keystorev4.New().Encrypt(sk, encryptionPassword)
-			if err != nil {
-				logger.Fatal("Failed to encrypt private key", zap.Error(err))
-			}
-
-			encryptedJSON, err := json.Marshal(encryptedData)
-			if err != nil {
-				logger.Fatal("Failed to marshal encrypted data to JSON", zap.Error(err))
-			}
-
-			err = os.WriteFile("encrypted_private_key.json", encryptedJSON, 0600)
-			if err != nil {
-				logger.Fatal("Failed to write encrypted private key to file", zap.Error(err))
-			}
-
-			logger.Info("private key encrypted and stored in encrypted_private_key.json")
+		if passwordFilePath != "" {
+			encryptAndSavePrivateKey(sk, passwordFilePath, logger)
```

Comment:
please use similar structure to what I described above.
one function that encrypts and returns either encrypted data or error, and then use the writeFile func to write the returned data.